src/db.rs

```diff
@@ -71,6 +71,77 @@ impl DB {
         Ok(())
     }
 
+    pub fn migrate<P: AsRef<Path>>(&mut self, path: P) -> Result<()> {
+        if !self.dirs.is_empty() {
+            // FIXME: just for testing
+            #[cfg(not(debug_assertions))]
+            return Err(anyhow!(
+                "To prevent conflicts, you can only migrate from z with an empty \
+                 zoxide database!"
+            ));
+        }
+
+        let z_db_file = File::open(path).with_context(|| anyhow!("could not open z db file"))?;
+        let reader = BufReader::new(z_db_file);
+
+        for (idx, read_line) in reader.lines().enumerate() {
+            let line_number = idx + 1;
+            let line = if let Ok(line) = read_line {
+                line
+            } else {
+                eprintln!("could not read line {}: {:?}", line_number, read_line);
```

Comment:
My "best effort" at "best effort" parsing. Might have missed some things; do let me know if that happens to be the case.